pkg/operator/apis/monitoring/v1/pod_config.go

```diff
@@ -382,6 +386,76 @@ func relabelingsForMetadata(keys map[string]struct{}) (res []*relabel.Config) {
 			TargetLabel:  "node",
 		})
 	}
+	if _, ok := keys["top_level_controller_name"]; ok {
+		res = append(res,
+			// Default to pod name
+			&relabel.Config{
+				Action:       relabel.Replace,
+				SourceLabels: prommodel.LabelNames{"__meta_kubernetes_pod_name"},
+				TargetLabel:  "top_level_controller_name",
+			},
+			// For non-ReplicaSet and non-Job controllers
+			&relabel.Config{
+				Action:       relabel.Replace,
+				SourceLabels: prommodel.LabelNames{"__meta_kubernetes_pod_controller_name"},
+				Regex:        relabel.MustNewRegexp("(.+)"),
+				TargetLabel:  "top_level_controller_name",
+				Replacement:  "$1",
+			},
+			// ReplicaSets will overwrite the label using the prefix if they came from Deployments
+			&relabel.Config{
+				Action:       relabel.Replace,
+				SourceLabels: prommodel.LabelNames{"__meta_kubernetes_pod_controller_kind", "__meta_kubernetes_pod_labelpresent_pod_template_hash", "__meta_kubernetes_pod_controller_name"},
+				Regex:        relabel.MustNewRegexp("ReplicaSet;true;(.+)-[a-z0-9]+"),
+				TargetLabel:  "top_level_controller_name",
+				Replacement:  "$1",
+			},
+			// CronJobs will match the pattern with a timestamp in minutes
+			&relabel.Config{
+				Action:       relabel.Replace,
+				SourceLabels: prommodel.LabelNames{"__meta_kubernetes_pod_controller_kind", "__meta_kubernetes_pod_controller_name"},
+				Regex:        relabel.MustNewRegexp("Job;(.+)-\\d{8,}$"),
+				TargetLabel:  "top_level_controller_name",
+				Replacement:  "$1",
+			},
+		)
+	}
+	if _, ok := keys["top_level_controller_type"]; ok {
+		res = append(res,
+			// Default to Pod
+			&relabel.Config{
+				Action:       relabel.Replace,
+				SourceLabels: prommodel.LabelNames{"__meta_kubernetes_pod_name"},
+				Regex:        relabel.MustNewRegexp("(.*)"),
+				TargetLabel:  "top_level_controller_type",
+				Replacement:  "Pod",
```

Comment:
Curious, why `pod` and not `unknown` (clear distinction that we can't / have bug in the workload discovery?) . Not a bad choice, just curious of pros & cons, can't see this explained in the design.

Same for controller_name 🤔